community/modules/compute/gke-job-template/main.tf

```diff
@@ -0,0 +1,58 @@
+/**
+  * Copyright 2023 Google LLC
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+locals {
+  should_set_resources = var.cpu_per_node != null
+  cpu_limit            = local.should_set_resources ? var.cpu_per_node : 0
+  cpu_request          = local.cpu_limit > 2 ? local.cpu_limit - 1 : "${local.cpu_limit * 1000 / 2 + 10}m"
```

Comment:
This seems very mysterious, would you mind describing why `cpu_request` is set with this value?

It seems the "default" here will be: 
cpu_per_node=null (default value)
shhould_set_resources=false
cpu_limit = 0
cpu_request = 0*1000/2 + 10 = 10m --> are you saying that by default jobs have a 10m deadline? 

From the docs I [saw](https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-resource-requests-and-limits), which do not account for batch as much and so can be wrong: 
```
CPU resources are defined in millicores. If your container needs two full cores to run, you would put the value “2000m”. If your container only needs ¼ of a core, you would put a value of “250m”.
```
Wouldn't it make sense that the cpu request is 1000 or 1010 by default? (and maybe exposed in case customers want to say they really just need a fraction of a cpu?)

Reply:
I have tried to address this by:
- fixing bug you point out about zero default
- Use actual GKE published allocatable resources as basis
- providing a user override
- better documenting the process/logic

These changes are self contained in https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/1234/commits/ee1bde830dc8a1805db78b3db6e76c1e5db18945.